js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx

```diff
@@ -93,8 +98,14 @@ type AdvancedSettingsProps = { isOIDC: boolean; isSAML: boolean };
 
 export const AdvancedSettings = ({ isOIDC, isSAML }: AdvancedSettingsProps) => {
   const { t } = useTranslation("identity-providers");
-  const { control } = useFormContext();
+  const { 
+    control,
+    register,
+    formState: { errors },
+  } = useFormContext<IdentityProviderRepresentation>();
   const [syncModeOpen, setSyncModeOpen] = useState(false);
+  const [filteredByClaim, setFilteredByClaim] = useState(control._formValues["config"]["filteredByClaim"]);
```

Comment:
Don't use "private" fields. You should replace this by a "watch"
```suggestion
  const filteredByClaim = useWatch({ control, name: "config.filteredByClaim", defaultValue: false});
```